### PR TITLE
fuzz: use number maker to produce numbers of a specific type.

### PR DESCRIPTION
**Pull request by Hijiao**
Signed-off-by: Max <xiahaijiao@hotmail.com>

## What have you changed? (mandatory)

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

As `make_u64()`, `make_i64()`, `make_i32()`, etc., are widely used in fuzzing, so I move all these functions into `number_maker.rs` file.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

pass  manual tests

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

If there is document change, please file a PR in ([docs](https://github.com/pingcap/docs) or [docs-cn](https://github.com/pingcap/docs-cn)) and add the PR number here.
no
## Does this PR affect tidb-ansible update? (mandatory)
no
If there is a configuration or metrics change, please file a PR in [tidb-ansible](https://github.com/pingcap/tidb-ansible), and add the PR number here.

### CODE REVIEW

File: fuzz/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (breeswish) [inline fuzz/src/lib.rs]: 2018. ditto to others.

File: fuzz/src/fuzz_targets/codec_number.rs [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
-where
-    I: Iterator<Item = u8>,
-{
-    let mut bytes = [0u8; 8];
-    for byte in &mut bytes {
-        *byte = iter.next().unwrap();
-    }
-    unsafe { mem::transmute(bytes) }
-}
+use tikv_fuzz::util::number_maker::{make_f64, make_i32, make_i64, make_u16, make_u32, make_u64};
```
Comment (breeswish) [inline fuzz/src/fuzz_targets/codec_number.rs]: I think it's more usual to just import `number_maker` and use `number_maker::make_f64` and so on in the rest.

File: fuzz/src/util/number_maker.rs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+make_numbers!{make_u64, 8, u64}
+make_numbers!{make_i64, 8, i64}
+make_numbers!{make_f64, 8, f64}
+make_numbers!{make_u32, 4, u32}
+make_numbers!{make_i32, 4, i32}
+make_numbers!{make_u16, 2, u16}
+make_numbers!{make_i8, 1, i8}
+
+pub fn make_bool<I>(iter: &mut I) -> bool
```
Comment (breeswish) [inline fuzz/src/util/number_maker.rs]: Is this function useful? Only true or false will be produced.

File: fuzz/src/fuzz_targets/codec_bytes.rs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+#![no_main]
+#[macro_use]
+extern crate libfuzzer_sys;
+extern crate tikv;
+
+fuzz_target!(|data: &[u8]| {
+    use tikv::util::codec;
+
+    let _ = codec::bytes::encode_bytes(data);
```
Comment (solotzg) [inline fuzz/src/fuzz_targets/codec_bytes.rs]: Have you considered to remove `let _ = `, or maybe you gonna use it later?

Comment (overvenus) [reply fuzz/src/fuzz_targets/codec_bytes.rs]: I think it is okay to ignore the returned value, as long as we follow rules listed in https://llvm.org/docs/LibFuzzer.html#id22

File: fuzz/src/util/number_maker.rs (hunk 3 above)
Comment (Hijiao) [reply fuzz/src/util/number_maker.rs]: yes, it is. This function is used to randomly produce bool value in my next fuzzing cases

File: fuzz/src/fuzz_targets/codec_number.rs (hunk 2 above)
Comment (Hijiao) [reply fuzz/src/fuzz_targets/codec_number.rs]: got it~

**Comment by Hijiao** [comment]
@overvenus @breeswish @solotzg PTAL

File: fuzz/src/fuzz_targets/codec_bytes.rs (hunk 4 above)
Comment (Hijiao) [reply fuzz/src/fuzz_targets/codec_bytes.rs]: @solotzg  The function result is useless at all, but  i use `let _ = `to handle the warning like this:

> warning: unused `std::result::Result` which must be used
   --> fuzz-helper/src/main.rs:108:13
    |
108 |             buff.encode_duration(&duration);
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(unused_must_use)] on by default
    = note: this `Result` may be an `Err` variant, which should be handled

**Pull request merged**